independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanDeployment.java

```diff
@@ -363,6 +372,20 @@ ClassInfo getQualifier(DotName name) {
         return qualifiers.get(name);
     }
 
+    Function<AnnotationInstance, Collection<AnnotationInstance>> getQualifierExtractor(DotName name) {
```

Comment:
Looking at the usage I wonder if it makes sense to return a function. `Collection<AnnotationInstance> getQualifiers(AnnotationInstance annotation)` would be sufficient and more convenient in most cases. Alternatively, we could add a convenient method like this:
```java
Collection<AnnotationInstance> getQualifiers(AnnotationInstance annotation) {
    return getQualifierExtractor(annotation.name()).apply(annotation);
}
```

Reply:
Ok, I will change that to the suggested method. You're right that it's easier to read and use.